### PR TITLE
Fixes hover tooltip position with non-origin div position

### PR DESCRIPTION
**Pull request by monfera**
Issue reported by @alexcjohnson is https://github.com/plotly/plotly.js/issues/1662

### CODE REVIEW

File: src/traces/sankey/plot.js [hunk 1]
```diff
@@ -125,13 +125,14 @@ module.exports = function plot(gd, calcData) {
 
     var linkHoverFollow = function(element, d) {
 
+        var rootBBox = gd.getBoundingClientRect();
```
Comment (etpinard) [inline src/traces/sankey/plot.js]: Can you use `fullLayout.width` and `fullLayout.height` or even `fullLayout._size` instead? Those `getBoundingClientRect` calls are expensive.

Comment (monfera) [inline src/traces/sankey/plot.js]: thanks, I'll check

Comment (monfera) [inline src/traces/sankey/plot.js]: ... maybe I could use help for a low-latency fix :-) the code needs to know the positions, not just the width/height - having said this, the speed impact is likely negligible as it's only triggered on one element at a time. Yet if there's a need to quickly switch, I could use suggestions for getting the right position without a bbox call.

Comment (alexcjohnson) [inline src/traces/sankey/plot.js]: this matches the pattern we've had to use elsewhere. It's possible that *both* `getBoundingClientRect` calls could be avoided by figuring out the position directly from the link/node element (or by caching it in the element's d3 data or something) but if you use `getBoundingClientRect` for one you'd better use it for both. That could be an optimization later, this is good for now.

File: src/traces/sankey/plot.js [hunk 2]
```diff
@@ -125,13 +125,14 @@ module.exports = function plot(gd, calcData) {
...
         var boundingBox = element.getBoundingClientRect();
         var hoverCenterX = boundingBox.left + boundingBox.width / 2;
         var hoverCenterY = boundingBox.top + boundingBox.height / 2;
 
         var tooltip = Fx.loneHover({
-            x: hoverCenterX + window.scrollX,
-            y: hoverCenterY + window.scrollY,
+            x: hoverCenterX - rootBBox.left + window.scrollX,
+            y: hoverCenterY - rootBBox.top + window.scrollY,
```
Comment (alexcjohnson) [inline src/traces/sankey/plot.js]: are the `window.scrollX/Y` shifts still necessary now that you're using a `getBoundingClientRect` differential?

**Pull request merged**